clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -692,6 +692,8 @@ protected void onJoinPrepare(int generation, String memberId) {
 
     @Override
     public void onLeavePrepare() {
+        log.debug("Executing onLeavePrepare with generation {} and memberId {}", generation(), memberId());
```

Comment:
Since generation may change by the hb in between this line and line 702 below, this entry may not have the exact number, maybe just read out from `generation()` and use that object?